indexd/fence_client.py

```diff
@@ -28,7 +30,11 @@ def get_signed_url_for_object(self, object_id, access_id):
         fence_server = self.url
         api_url = fence_server.rstrip("/") + "/data/download/"
         url = api_url + object_id
-        if access_id:
+        if access_id is not None and access_id != "":
+            if access_id not in SUPPORTED_PROTOCOLS:
+                raise UserError(
+                    "The specified protocol {} is not supported".format(access_id)
+                )
```

Comment:
The thing with this is that, given the incorrect protocol, fence doesn't return an error that indexd can read as far as I know. If I can remember correctly it either gives me some kind of 404 instead of a 400.
ex: _{"msg":"No document with id:blah1234 with access_id:s2","status_code":404}_ which i thought was misleading. I dont know. What do you suggest?

Reply:
i think it's fine to return an error that lets the user know either the guid is not found or the access_id is not found. looking at the fence code, it looks like it returns a 400